presto-expressions/src/main/java/com/facebook/presto/expressions/translator/FunctionTranslator.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.expressions.translator;
+
+import com.facebook.presto.spi.function.FunctionMetadata;
+import com.google.common.collect.ImmutableMap;
+
+import java.lang.invoke.MethodHandle;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+public class FunctionTranslator<T>
+{
+    private final Map<FunctionMetadata, MethodHandle> functionMapping;
+
+    public static <T> FunctionTranslator<T> buildFunctionTranslator(Set<Class<?>> translatorContainers)
+    {
+        ImmutableMap.Builder<FunctionMetadata, MethodHandle> functionMappingBuilder = new ImmutableMap.Builder<>();
+        translatorContainers.stream()
+                .map(TranslatorAnnotationParser::parseFunctionDefinitions)
+                .forEach(functionMappingBuilder::putAll);
+        return new FunctionTranslator<>(functionMappingBuilder.build());
+    }
+
+    private FunctionTranslator(Map<FunctionMetadata, MethodHandle> functionMapping)
+    {
+        this.functionMapping = functionMapping;
+    }
+
+    public boolean isFunctionPresent(FunctionMetadata functionMetadata)
+    {
+        return functionMapping.containsKey(functionMetadata);
+    }
+
+    public Optional<T> translate(FunctionMetadata functionMetadata, List<TranslatedExpression<T>> translatedArguments)
```

Comment:
I think this should return `TranslatedExpression<T>`. Could you add test to cover this?